### ISSUE TITLE
RenderButtonCustom.OnClickFunctionName only allows numeric parameters

### ISSUE THREAD

**Issue by manuel3108** [nopSolutions/nopCommerce#6028]
nopCommerce version: 4.40.x and above

Steps to reproduce the problem: 
Reference: https://github.com/nopSolutions/nopCommerce/blob/cf1cdbe9ccc7a5dfa44f3617e6ed92583e353f84/src/Presentation/Nop.Web/Areas/Admin/Views/Currency/List.cshtml#L108-L116
1. Create a new DataTable
2. And create a columproperty rendering an RenderButtonCustom
3. Set a OnClickFunctionName to whatever you want
4. Make sure that your parameter of ColumnProperty refers to a non-numeric value, i.e.  a string.
5. Open the change in your browser and have a look at your console. Depending on your string content, you will see an error message there or not. But if you navigate to the source code of the button click, you will see that your string is supposed to be an object. You will probably get an error upon calling your custom button.

Workaround (really messy):
Render your column like this (take care on the extra braces in OnClickFunctionName ):
```c#
                               new ColumnProperty(nameof(LocaleResourceModel.Id))
                                    {
                                        Title = T("myresource").Text,
                                        Width = "100",
                                        ClassName =  NopColumnClassDefaults.Button,
                                        Render = new RenderButtonCustom(NopButtonClassDefaults.Default, T("Plugins.Misc.FancyPdf.InsertIntoEditor").Text)
                                        {
                                            OnClickFunctionName =  "copyToTinyMce(this);"
                                        }
                                    }
```

and then get the value your parameter something like this in JS
```js
                          function copyToTinyMce(element) {
                                let text = $(element).parent().parent().find("td:first-child").text();
                                // do something
                            }
```

**Comment by DmitriyKulagin** [comment]
@manuel3108 The data string identifier is passed to the OnClickFunctionName function. With this data in the future, you can do whatever you want at the service level. However, we did not set ourselves the goal of supporting something else, much less navigation through the DOM model at the view level (this follows from your example). If you want to change this behavior, it is better to [change the generation of the OnClickFunctionName function handler itself](https://github.com/nopSolutions/nopCommerce/blob/d9c79c6b61ad9c5cdfc544f2b8bec16c3ceb13f3/src/Presentation/Nop.Web/Areas/Admin/Views/Shared/_Table.Definition.cshtml#L362-L366).

**Comment by manuel3108** [comment]
@DmitriyKulagin I think there was a misunderstanding from my example.
I will try to clarify this better.

Lets assume the following Column. We have a column called UpdateUser data, which renders a button with the same text shown below. Because we are communicating with an external API, we need the UserGuid, which should be the data-parameter for this function.

So lets imagine the following column definition
```c#
new ColumnProperty(nameof(MyUser.GuidFormattedAsString))
  {
      Title = T("Plugins.Misc.MyPlugin.UpdateUserData").Text,
      Width = "100",
      ClassName =  NopColumnClassDefaults.Button,
      Render = new RenderButtonCustom(NopButtonClassDefaults.Default, T("Plugins.Misc.MyPlugin.UpdateUserData").Text)
      {
          OnClickFunctionName =  "doSomethingWithTheUid"
      }
  }
```

and the following JS code
```js
function doSomethingWithTheUid(myString) {
  // call an api with the user uid
}
```

So now let's get to an example. Let's assume the user guid of is `a941b124-5740-41b8-8b62-27bd7ee06596`. At the moment being, nopCommerce would create a function call like this:
```js
doSomethingWithTheUid(a941b124-5740-41b8-8b62-27bd7ee06596)
```
This of course is invalid JS.

The expected call should instead be
```js
doSomethingWithTheUid("a941b124-5740-41b8-8b62-27bd7ee06596")
```


----
In order to easily identify the problem, paste the following into your developer console:
```js
function doSomething(asd){console.log(asd)}
doSomething(a941b124-5740-41b8-8b62-27bd7ee06596)
```
which will give you `Uncaught SyntaxError: identifier starts immediately after numeric literal`

----
I agree, my previous explanation was bad.
As I cannot re-open the issue, please feel free to re-open the issue

**Comment by DmitriyKulagin** [comment]
If your suggestion is to pass a string as a parameter to a function, then this might actually be useful.

**Comment by DmitriyKulagin** [comment]
Closed #6028

**Comment by manuel3108** [comment]
Yes it was. Thanks for the fix
